internal/types/resource/resource.go

```diff
@@ -90,3 +93,17 @@ var Map = map[string]Type{
 	CredentialStore.String():   CredentialStore,
 	CredentialLibrary.String(): CredentialLibrary,
 }
+
+type MinimalInfo struct {
+	PublicId string
+	ScopeId  string
+	UserId   string
+}
+
+type MinimalInfoRepo interface {
+	// Fetches resource IDs for the given scopes. Note that it is assumed that
+	// the set of scopes is either length 1 or is recursive; that is, the reason
+	// we would have multiple scopes is because there is a parent or set of
+	// parents and any children.
```

Comment:
This note and its implications are a little tough to grok. When we chatted about it it seemed that main thing that an implementer needs to address is the different between fetching for all scopes vs fetching for a specific scope. This could be expressed a few ways such as having separate methods (i.e. `FetchIdsForScopes`, `FetchIds`), or using an option pattern ie `FetchIds(context.Context, Option...)` or clarifying the behavior if `scopesIds []string` is empty.